src/services/db/GitFileSystemService.ts

```diff
@@ -337,6 +338,86 @@ export default class GitFileSystemService {
     )
   }
 
+  /**
+   * Wrapper over `git diff --name-only` that also creates a local tracking branch for `base` and/or `head` if they do not exist.
```

Comment:
local tracking branch means? if for example, master does not exist, it creates a branch called master?

btw base/head are symbolic references than branch names right